core/arch/x86/encode.c

```diff
@@ -2834,6 +2851,19 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     CLIENT_ASSERT(!di.vex_encoded || !di.evex_encoded,
                   "instr_encode error: flags can't be both vex and evex.");
 
+    if (di.evex_encoded) {
+        /* The upper half of the flags field is the evex tuple type. */
+        di.tuple_type = (byte)(info->flags >> 16);
+    }
+    if (di.vex_encoded || di.evex_encoded) {
+        if (TEST(OPCODE_MODRM, info->opcode)) {
+            /* The prefix is needed for evex when encoding displacements, in order
+             * to determine the compressed scale factor.
```

Comment:
This makes it sound like this code is not needed for anything else beyond displacements.

Reply:
True. I think the comment is not needed at all, I've removed it.